src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1370,6 +1370,15 @@ BoundExpression bindDefaultArgument(SyntaxNode syntax, ParameterSymbol parameter
                     var memberName = containingMember.GetMemberCallerName();
                     defaultValue = new BoundLiteral(syntax, ConstantValue.Create(memberName), Compilation.GetSpecialType(SpecialType.System_String)) { WasCompilerGenerated = true };
                 }
+                else if (callerSourceLocation is object && parameter.CallerArgumentExpressionParameterIndex != -1)
+                {
+                    // - Test extension methods.
+                    // - Test combining different attributes (e.g, both CallerArgumentExpression and MemberCallerName).
+                    // - Test having the CallerArgumentExpression tied to an optional parameter that doesn't exist in callsite.
+                    // - Test having a non-existent expression.
+                    var argument = arguments[parameter.CallerArgumentExpressionParameterIndex];
```

Comment:
It might be wrong to depend on the index. Keeping that temporarily until I get the "basic" of the feature working, then will write tests that fail with this and think of another way.